generator/internal/dart/darttemplate.go

```diff
@@ -150,14 +169,64 @@ func annotateModel(model *api.API, options map[string]string) (*modelAnnotations
 			}
 			return ""
 		}(),
-		DocLines: formatDocComments(model.Description, model.State),
+		DocLines:            formatDocComments(model.Description, model.State),
+		HasDependencies:     len(deps) > 0,
+		PackageDependencies: deps,
+		Imports:             calculateImports(importMap),
 	}
 
 	model.Codec = ann
 	return ann, nil
 }
 
-func annotateService(s *api.Service, state *api.APIState) {
+// Calculate package dependencies based on `package:` imports.
+func calculateDependencies(importMap map[string]*dartImport) []packageDependency {
+	var deps []packageDependency
+
+	for _, imp := range importMap {
+		if strings.HasPrefix(imp.DartImport, "package:") {
+			name := strings.TrimPrefix(imp.DartImport, "package:")
+			name = strings.Split(name, "/")[0]
+
+			deps = append(deps, packageDependency{Name: name, Constraint: "any"})
+		}
+	}
+
+	sort.SliceStable(deps, func(i, j int) bool {
+		return deps[i].Name < deps[j].Name
+	})
+
+	return deps
+}
+
+func calculateImports(importMap map[string]*dartImport) []string {
+	var dartImports []string
+	for _, imp := range importMap {
+		dartImports = append(dartImports, imp.DartImport)
+	}
+	sort.Strings(dartImports)
+
+	previousImportType := ""
+	var imports []string
+
+	for _, imp := range dartImports {
+		// Emit a blank line when changing between 'dart:' and 'package:' imports.
+		importType := strings.Split(imp, ":")[0]
+		if previousImportType != "" && previousImportType != importType {
+			imports = append(imports, "")
+		}
+		previousImportType = importType
+
+		imports = append(imports, fmt.Sprintf("import '%s';", imp))
+	}
+
+	return imports
+}
+
+func annotateService(s *api.Service, state *api.APIState, importMap map[string]*dartImport) {
+	// Require package:http when generating services.
+	importMap[httpImport.Package] = httpImport
```

Comment:
FWIW, in the Rust codec we moved from calculating the deps as we annotated the model to using a separate pass (see #904).  If you find that convincing, I would encourage you to do the same here.  At least keep it in mind as the complexity grows.

Reply:
Good to know; I'll probably refactor when I look to create deps between generated packages. At some point we'll also need to add code to use library prefixes when and if necessary to avoid name conflicts w/ generated symbols.